### ISSUE TITLE
OpenShift on OpenStack deployment fails on "Apply the SDN config" step

### ISSUE THREAD

**Issue by tzumainn** [openshift/openshift-ansible#7532]
#### Description

Deploying OpenShift on OpenStack off of master results in an error when running the "Apply the SDN config" step.

##### Version
```
ansible 2.4.3.0
  config file = /home/stack/ansible.cfg
  configured module search path = [u'/home/stack/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible
  python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
```

openshift-ansible-3.10.0-0.4.0-117-g6589945

##### Steps To Reproduce
1. Setup environment as described in https://github.com/openshift/openshift-ansible/blob/master/playbooks/openstack/README.md
2. Run 'ansible-playbook --user openshift -i openshift-ansible/playbooks/openstack/inventory.py -i inventory --private-key ~/.ssh/openshift openshift-ansible/playbooks/openstack/openshift-cluster/provision_install.yml '

##### Expected Results
Expect the deployment to finish without failing.

##### Observed Results
This error happens consistently:

```
TASK [openshift_sdn : Apply the SDN config] ********************************************************************************
Wednesday 14 March 2018  15:06:10 +0100 (0:00:01.227)       0:12:01.038 *******
fatal: [master-0.openshift.example.com]: FAILED! => {"changed": true, "cmd": "oc apply -f /tmp/console-ansible-wjXoJQ", "delta": "0:00:00.430827", "end": "2018-03-14 14:06:11.399480", "msg": "non-zero return code", "rc": 1, "start": "2018-03-14 14:06:10.968653", "stderr": "unable to recognize \"/tmp/console-ansible-wjXoJQ/sdn-ovs.yaml\": no matches for apps/, Kind=DaemonSet\nunable to recognize \"/tmp/console-ansible-wjXoJQ/sdn.yaml\": no matches for apps/, Kind=DaemonSet", "stderr_lines": ["unable to recognize \"/tmp/console-ansible-wjXoJQ/sdn-ovs.yaml\": no matches for apps/, Kind=DaemonSet", "unable to recognize \"/tmp/console-ansible-wjXoJQ/sdn.yaml\": no matches for apps/, Kind=DaemonSet"], "stdout": "imagestreamtag \"node:v3.9\" created\nserviceaccount \"sdn\" created\nclusterrolebinding \"sdn-cluster-reader\" created\nclusterrolebinding \"sdn-reader\" created", "stdout_lines": ["imagestreamtag \"node:v3.9\" created", "serviceaccount \"sdn\" created", "clusterrolebinding \"sdn-cluster-reader\" created", "clusterrolebinding \"sdn-reader\" created"]}
```

##### Additional Information

Deploying OpenShift on OpenStack off of master was working fairly recently; it started failing when I updated my master checkout on Monday. I believe this commit may be related?

https://github.com/openshift/openshift-ansible/commit/b17728d54294b8b273c636ba85303cfe067353f7

**Comment by tomassedovic** [comment]
So, that commit introduced the following file:

https://github.com/openshift/openshift-ansible/blob/0b40b5d5f383f321230b2a357d1cda83ef6d393e/roles/openshift_sdn/files/sdn.yaml#L2

(and sdn-ovs.yaml) and they both have `apiVersion: apps/v1` in them. Now the version of OpenShift Origin that gets currently installed is 3.7 and it (or rather the Kubernetes in bundles) does not understand that version declaration. Saying `apiVersion: extensions/v1beta1` instead would have worked.

However, there are going to be more changes like that, and basically, the master checkout of openshift-ansible will not work with old versions of OpenShift any more.

So the best workaround so far is to specify this in your inventory:

    openshift_repos_enable_testing: true

That will use the testing repos which ship an alpha build of 3.9. Once 3.9 is released in the main CentOS repos, this error should just go away.

**Comment by flipkill1985** [comment]
What do you mean with inventory?

**Comment by tomassedovic** [comment]
Yes, that's the one (unless you're passing a different file in explicitly).

Where did you put it? It should be under `[OSEv3:vars]`.

Do you have `openshift_release` or `openshift_pkg_version` set to anything? I think it should be either empty or set to `v3.9`. If you happen to have e.g. `v3.7`, you'll run into the same problem: the openshift packages being too old.

Could you SSH into one of the deployed nodes and run `oc version`? Paste the result here please.

**Comment by flipkill1985** [comment]
Thats my playbook, The Master and Node are on the Same Server i have only one Server.

`[OSEv3:children]
masters
etcd
nodes

[OSEv3:vars]
openshift_master_default_subdomain=apps.ex.peter.es
ansible_ssh_user=root
ansible_become=yes
openshift_master_cluster_method=native
openshift_master_cluster_hostname=sp-mohs02.ex.peter.es
openshift_master_cluster_public_hostname=openshift.ex.peter.es
deployment_type=origin

openshift_release=v3.7
openshift_repos_enable_testing=true

os_sdn_network_plugin_name='redhat/openshift-ovs-multitenant'
openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/htpasswd'}]
openshift_docker_options='--selinux-enabled --insecure-registry 172.30.0.0/16'
openshift_router_selector='region=infra'
openshift_registry_selector='region=infra'
openshift_master_api_port=443
openshift_master_console_port=443
openshift_disable_check=memory_availability,disk_availability,package_version

[nodes]
master openshift_schedulable=True ansible_connection=local ansible_become=yes

[masters]
master ansible_connection=local ansible_become=yes

[etcd]
master ansible_connection=local ansible_become=yes`

**Comment by tomassedovic** [comment]
Ah. You must replace `openshift_release=v3.7` with `openshift_release=v3.9`. OpenShift 3.7 no longer works with the master branch of openshift-ansible.

If you want to keep using 3.7, you should check out the appropriate branch or tag. `git checkout release-3.7` should work, but I have not tested it.

**Comment by flipkill1985** [comment]
ok, its ok for me to use 3.9 but now i get this error:
```
fatal: [master]: FAILED! => {
    "attempts": 120,
    "changed": false,
    "cmd": [
        "curl",
        "--silent",
        "--tlsv1.2",
        "--cacert",
        "/etc/origin/master/ca-bundle.crt",
        "https://sp-mohs02.os.mohs.es:8443/healthz/ready"
    ],
    "delta": "0:00:00.092607",
    "end": "2018-03-22 13:36:38.974384",
    "invocation": {
        "module_args": {
            "_raw_params": "curl --silent --tlsv1.2 --cacert /etc/origin/master/ca-bundle.crt https://sp-mohs02.os.mohs.es:8443/healthz/ready",
            "_uses_shell": false,
            "chdir": null,
            "creates": null,
            "executable": null,
            "removes": null,
            "stdin": null,
            "warn": false
        }
    },
    "msg": "non-zero return code",
    "rc": 35,
    "start": "2018-03-22 13:36:38.881777",
    "stderr": "",
    "stderr_lines": [],
    "stdout": "",
    "stdout_lines": []
```

```
origin-master-api.service - Atomic OpenShift Master API
   Loaded: loaded (/usr/lib/systemd/system/origin-master-api.service; enabled; vendor preset: disabled)
   Active: active (running) since Thu 2018-03-22 13:22:44 CET; 3min 11s ago
     Docs: https://github.com/openshift/origin
 Main PID: 15902 (openshift)
   CGroup: /system.slice/origin-master-api.service
           └─15902 /usr/bin/openshift start master api --config=/etc/origin/master/master-config.yaml --loglevel=2 --listen=https://0.0.0.0:8443 --master=https://sp-mohs02.os.mohs.es:8443

Mar 22 13:25:00 sp-mohs02.os.mohs.es atomic-openshift-master-api[15902]: I0322 13:25:00.788397   15902 logs.go:41] http: TLS handshake error from 212.75.37.96:54504: remote error: tls: bad certificate
Mar 22 13:25:02 sp-mohs02.os.mohs.es atomic-openshift-master-api[15902]: I0322 13:25:01.998047   15902 logs.go:41] http: TLS handshake error from 212.75.37.96:54508: remote error: tls: bad certificate
Mar 22 13:25:03 sp-mohs02.os.mohs.es atomic-openshift-master-api[15902]: I0322 13:25:03.213211   15902 logs.go:41] http: TLS handshake error from 212.75.37.96:54512: remote error: tls: bad certificate
Mar 22 13:25:04 sp-mohs02.os.mohs.es atomic-openshift-master-api[15902]: I0322 13:25:04.428621   15902 logs.go:41] http: TLS handshake error from 212.75.37.96:54516: remote error: tls: bad certificate
Mar 22 13:25:05 sp-mohs02.os.mohs.es atomic-openshift-master-api[15902]: I0322 13:25:05.679075   15902 logs.go:41] http: TLS handshake error from 212.75.37.96:54520: remote error: tls: bad certificate
Mar 22 13:25:06 sp-mohs02.os.mohs.es atomic-openshift-master-api[15902]: I0322 13:25:06.860232   15902 logs.go:41] http: TLS handshake error from 212.75.37.96:54524: remote error: tls: bad certificate
Mar 22 13:25:08 sp-mohs02.os.mohs.es atomic-openshift-master-api[15902]: I0322 13:25:08.090551   15902 logs.go:41] http: TLS handshake error from 212.75.37.96:54528: remote error: tls: bad certificate
Mar 22 13:25:09 sp-mohs02.os.mohs.es atomic-openshift-master-api[15902]: I0322 13:25:09.291960   15902 logs.go:41] http: TLS handshake error from 212.75.37.96:54532: remote error: tls: bad certificate
Mar 22 13:25:10 sp-mohs02.os.mohs.es atomic-openshift-master-api[15902]: I0322 13:25:10.506321   15902 logs.go:41] http: TLS handshake error from 212.75.37.96:54536: remote error: tls: bad certificate
Mar 22 13:25:10 sp-mohs02.os.mohs.es atomic-openshift-master-api[15902]: I0322 13:25:10.729394   15902 logs.go:41] http: TLS handshake error from 212.75.37.96:54540: remote error: tls: bad certificate
```

**Comment by tomassedovic** [comment]
Yeah, I think there are multiple issues with deploying origin on master now :-(. But the issue logged here should be resolved with 3.9.
